### PR TITLE
ampt: add new package

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ampt/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+       nuclear collisions at relativistic energies."""
+
+    homepage = "http://myweb.ecu.edu/linz/ampt/"
+    url      = "http://myweb.ecu.edu/linz/ampt/ampt-v1.26t9b-v2.26t9b.zip"
+
+    maintainers = ['vvolkl']
+
+    tags = ['hep']
+
+    version('v2.26t9', sha256='d90e2e5a88f93baec69bc56d67ed8e35dc243fd1e5104e551e8d7e69fd09716a')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/ampt/package.py]: Would it be possible to simplify the version scheme used? For instance omit the "v" and have "t9" after a dash? Also what are the two versions in the URL (there's also `v1.26t9b`)?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/ampt/package.py (hunk 1 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/ampt/package.py]: I'd prefer to keep the version as is, for continuity with previous non-spack builds (https://lcginfo.cern.ch/pkgver/ampt/2.26t9b_atlas/) Although I just notice that I still have to add the atlas patch.
EDIT: did remove the `v`
EDIT: and added the dash, it really is easier to read

According to the website, the v1 series uses the default model, the v2 series a model with string melting, but they are always packaged together, so I think just using the v2 as version string is ok.

File: var/spack/repos/builtin/packages/ampt/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    homepage = "http://myweb.ecu.edu/linz/ampt/"
+    url      = "http://myweb.ecu.edu/linz/ampt/ampt-v1.26t9b-v2.26t9b.zip"
+
+    maintainers = ['vvolkl']
+
+    tags = ['hep']
+
+    patch('https://gitlab.cern.ch/sft/lcgcmake/-/raw/master/generators/patches/ampt-2.26t9b_atlas.patch', level=0,
+          sha256='7a9a4f175f84dc3021301dae5d48adab1fc714fccf44ec17128a3ba1608bff4c',
+          when='@2.26t9b_atlas')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/ampt/package.py]:
```suggestion
          when='@2.26-t9b_atlas')
```

**Review by alalazo — CHANGES_REQUESTED** [review]
I think I spotted a typo, but otherwise LGTM

File: var/spack/repos/builtin/packages/ampt/package.py (hunk 2 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/ampt/package.py]: Good catch, thanks!

**Review by alalazo — APPROVED** [review]

**Pull request merged**